### PR TITLE
add Windows console startup description

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

File: spec/windows.dd [hunk 1]
```diff
@@ -8,6 +8,55 @@ $(P This covers Windows programming with 32 bit OMF, 32 bit Mscoff,
...
+                {
+                    return main(argc, argv);
+                }
+            }
+        }
+    }
+    ---
+    $(P The `main` function in it, because it is marked as `extern(C)`, is the entry point of a
+    C console program. The executable starts up as a C executable, with the C runtime library
+    initializing and then the C main() function is called.)
```
Comment (RazvanN7) [inline spec/windows.dd:44]:
```suggestion
    initialization and then the C main() function is called.)
```

File: spec/windows.dd [hunk 2]
```diff
@@ -8,6 +8,55 @@ $(P This covers Windows programming with 32 bit OMF, 32 bit Mscoff,
...
+    $(P The C runtime library will runs any D functions tagged with `pragma(crt_constructor)` as
```
Comment (RazvanN7) [inline spec/windows.dd:45]:
```suggestion
    $(P The C runtime library runs any D functions tagged with `pragma(crt_constructor)` as
```

**Review by RazvanN7 — DISMISSED** [review]

File: spec/windows.dd [hunk 3]
```diff
@@ -8,6 +8,55 @@ $(P This covers Windows programming with 32 bit OMF, 32 bit Mscoff,
...
+            int _d_run_main(int argc, char **argv, void* mainFunc);
+
+            int _Dmain(char[][] args);
+
+            int main(int argc, char **argv)
+            {
+                return _d_run_main(argc, argv, &_Dmain);
+            }
+
+            // Solaris, for unknown reasons, requires both a main() and an _main()
```
Comment (ibuclaw) [inline spec/windows.dd:31]: Has anyone tried asking Solaris?

**Comment by rikkimax** [comment]
Ping

**Review by thewilsonator — APPROVED** [review]

**Pull request merged**